### PR TITLE
IGNITE-12583 : refactored JdbcThinBulkLoadAbstractSelfTest into a parametrized one

### PR DESCRIPTION
**Pull request by Vladsz83**
The following test are combined into one parametrized JdbcThinBulkLoadSelfTest:

JdbcThinBulkLoadAtomicPartitionedNearSelfTest
JdbcThinBulkLoadAtomicPartitionedSelfTest
JdbcThinBulkLoadAtomicReplicatedSelfTest
JdbcThinBulkLoadTransactionalPartitionedNearSelfTest
JdbcThinBulkLoadTransactionalPartitionedSelfTest
JdbcThinBulkLoadTransactionalReplicatedSelfTest

### CODE REVIEW

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java [hunk 1]
```diff
@@ -574,9 +602,9 @@ public void testCreateAndBulkLoadTable() throws SQLException {
             " (id int primary key, age int, firstName varchar(30), lastName varchar(30))");
 
         int updatesCnt = stmt.executeUpdate(
-            "copy from '" + BULKLOAD_TWO_LINES_CSV_FILE + "' into " + tblName +
-                "(_key, age, firstName, lastName)" +
-                " format csv");
+                "copy from '" + BULKLOAD_TWO_LINES_CSV_FILE + "' into " + tblName +
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: Changes unrelated. Please, revert.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java [hunk 2]
```diff
@@ -43,15 +29,30 @@
 import org.apache.ignite.lang.IgniteClosure;
 import org.apache.ignite.testframework.GridTestUtils;
 import org.junit.Test;
+import org.junit.runner.RunWith;
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: Changes unrelated. Please, revert.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java [hunk 3]
```diff
@@ -30,6 +25,13 @@
 import org.apache.ignite.internal.binary.BinaryMarshaller;
 import org.apache.ignite.internal.util.typedef.F;
 
+import java.io.Serializable;
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java]: Changes unrelated. Please, revert.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/suite/IgniteJdbcDriverTestSuite.java [hunk 4]
```diff
@@ -17,225 +17,153 @@
...
-
-    JdbcThinLocalQueriesSelfTest.class,
-
-    // Various commands.
-    JdbcThinWalModeChangeSelfTest.class,
-    JdbcThinAuthenticateConnectionSelfTest.class,
-
-    JdbcThinPreparedStatementLeakTest.class,
-    JdbcThinTransactionsLeaksMvccTest.class,
+        JdbcConnectionSelfTest.class,
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/suite/IgniteJdbcDriverTestSuite.java]: Changes unrelated. Please, revert.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java [hunk 5]
```diff
@@ -164,6 +165,19 @@ final CacheConfiguration binaryCacheConfig() {
...
 
+    /** Clean tables to avoid DDL errors in repeated test runs.  */
+    protected void dropTables() throws SQLException {
+        List<List<?>> tablesLeft = execute(conn, "select CACHE_NAME, TABLE_NAME, SCHEMA_NAME from SYS.TABLES");
+        for( List<?> resultRecord : tablesLeft ){
+            String cacheName = resultRecord.get(0).toString();
+            if(cacheName.startsWith("SQL_")){
+                String tableName = resultRecord.get(1).toString();
+                String schemaName = resultRecord.get(2).toString();
+                execute(conn, "drop table "+schemaName+".\""+tableName+"\"");
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java]: empty line required.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java [hunk 6]
```diff
@@ -164,6 +165,19 @@ final CacheConfiguration binaryCacheConfig() {
         return cache;
     }
 
+    /** Clean tables to avoid DDL errors in repeated test runs.  */
+    protected void dropTables() throws SQLException {
+        List<List<?>> tablesLeft = execute(conn, "select CACHE_NAME, TABLE_NAME, SCHEMA_NAME from SYS.TABLES");
+        for( List<?> resultRecord : tablesLeft ){
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java]: empty line required.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java [hunk 7]
```diff
@@ -89,6 +94,31 @@
...
+
+    /** Parametrized run param : near mode */
+    @Parameterized.Parameter(2)
+    public Boolean isNear;
+
+    /** Test run configurations: Cache mode, atomicity type, is near */
+    @Parameterized.Parameters
+    public static Collection<Object[]> runConfig() {
+        return Arrays.asList(new Object[][] {
+            { PARTITIONED, CacheAtomicityMode.ATOMIC, true },
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: We should use static import in both cases

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java [hunk 8]
```diff
@@ -154,21 +184,21 @@ private CacheConfiguration cacheConfigWithQueryEntity() {
      *
      * @return true if we are testing near cache.
      */
-    protected abstract boolean nearCache();
+    protected boolean nearCache(){ return isNear; };
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: Can we also remove this methods and use variables?

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java (hunk 5 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java]: Removed at all. Simplier approach involved: dropping table on the place: "drop table..."

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java (hunk 6 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinAbstractDmlStatementSelfTest.java]: Removed at all.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java (hunk 7 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: Fixed

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java (hunk 8 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: yep

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java [hunk 9]
```diff
@@ -89,6 +96,31 @@
     /** JDBC statement. */
     private Statement stmt;
 
+    /** Parametrized run param : cacheMode */
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: . in the end of the comment required.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java [hunk 10]
```diff
@@ -89,6 +96,31 @@
...
+    @Parameterized.Parameter(0)
+    public CacheMode cacheMode;
+
+    /** Parametrized run param : atomicity */
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: . in the end of the comment required.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java [hunk 11]
```diff
@@ -89,6 +96,31 @@
...
+    @Parameterized.Parameter(1)
+    public CacheAtomicityMode atomicityMode;
+
+    /** Parametrized run param : near mode */
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: . in the end of the comment required.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java [hunk 12]
```diff
@@ -89,6 +96,31 @@
...
+    @Parameterized.Parameter(2)
+    public Boolean isNear;
+
+    /** Test run configurations: Cache mode, atomicity type, is near */
```
Comment (nizhikov) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: . in the end of the comment required.

Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: fixed.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java (hunk 11 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: fixed.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java (hunk 10 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: fixed.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java (hunk 9 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBulkLoadSelfTest.java]: fixed.

**Pull request merged**